### ISSUE TITLE
Calculating money with rationals maybe wrong.

### ISSUE THREAD

**Issue by luftluft** [factor/factor#1786]
`IN: scratchpad auto-use 2.345 money.
`
Outputs $2.35.0
It should be $2.35 instead.

"cents >integer" could fix it.

`: dollars/cents ( dollars -- dollars cents )
    100 * 100 /mod round >integer ;
`

**Comment by jonenst** [comment]
In factor, 2.345 is not a rational, it is a float. A rational is either an integer or or ratio. It is an exact value,vs float which are approximate values.

Looking at the rest of the vocabulary, it looks like it was designed to be used like this:
```factor
DECIMAL: 2.345 money. ! $2.35
```

Not sure about DECIMAL: which conflicts with extra:decimals ...

**Comment by luftluft** [comment]
Thank you very much, Jon!

**Comment by mrjbq7** [comment]
Seems like an easy patch adding ``>integer`` to make ``money>string`` work with floats?  It seems to work with integers and ratios already. 

Note: it doesn't work with ``extra/decimals``.

**Comment by mrjbq7** [comment]
You have to admit this is funny looking:

```factor
IN: scratchpad 1.0 money.
$1.0.0
```

**Comment by mrjbq7** [comment]
Fixed in 9a34af7381b973e3e1591e54e6cfa4ef582994b3.

**Comment by jonenst** [comment]
Well doing money computations with floats is a Bad Idea™ . You will get subtle bugs with rounding issues if you are not extra careful. Maybe that's why the original design didn't support them.
